lib/mysqlDaoQueryHelper.js

```diff
@@ -58,6 +58,23 @@ class MysqlDaoQueryHelper {
     }
 
     /**
+     * @param {object} upsertArgs
+     * @returns {string}
+     */
+    upsert(upsertArgs) {
+        const cleanValues = this._cleanAndMapValues(upsertArgs);
+        let insert =
+            Squel.useFlavour('mysql').insert().into(this._tableName)
+                .setFields(cleanValues, {dontQuote: true});
+
+        Object.keys(cleanValues).forEach(property => {
+            insert.onDupUpdate(property, cleanValues[property], {dontQuote: true});
+        });
```

Comment:
Yeah so this would take the other values. You would also do the parsing of the "function" values here, too (similar to mysqlDaoQueryHelper's `_queryObjectToSql` method)